nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -189,9 +189,14 @@ let
           mkdir /boot/grub
           echo '(hd0) /dev/vda' > /boot/grub/device.map
 
-          # Install GRUB and generate the GRUB boot menu.
+          # Install bootloader
           touch /etc/NIXOS
           mkdir -p /nix/var/nix/profiles
+          ln -s ${config.system.build.toplevel} /nix/var/nix/profiles/system-1-link
+          ln -s /nix/var/nix/profiles/system-1-link /nix/var/nix/profiles/system
```

Comment:
This is not done by the `switch-to-configuration` script?

Reply:
Nope. Normally, this is done by `nixos-rebuild` which uses `nix-env -p /nix/var/nix/profile/system --set ...` before running `switch-to-configuration`.